primitives/runtime/src/bounded/weak_bounded_vec.rs

```diff
@@ -34,9 +39,69 @@ use sp_std::{marker::PhantomData, prelude::*};
 ///
 /// The length of the vec is not strictly bounded. Decoding a vec with more element that the bound
 /// is accepted, and some method allow to bypass the restriction with warnings.
+#[cfg_attr(feature = "std", derive(Serialize), serde(transparent))]
 #[derive(Encode, scale_info::TypeInfo)]
 #[scale_info(skip_type_params(S))]
-pub struct WeakBoundedVec<T, S>(pub(super) Vec<T>, PhantomData<S>);
+pub struct WeakBoundedVec<T, S>(
+	pub(super) Vec<T>,
```

Comment:
Hmm, but that would be weird, since deserializing as a `BoundedVec` will check whether the number of elements is within bounds, which is not what we want here.